app/uk/gov/hmrc/plasticpackagingtax/registration/forms/contact/Address.scala

```diff
@@ -70,10 +80,18 @@ object Address extends CommonFormValidators {
     townOrCity -> text()
       .verifying(emptyError(townOrCity), isNonEmpty)
       .verifying(notValidError(townOrCity), validateAddressField(addressFieldMaxSize)),
-    postCode -> text()
-      .transform[String](postCode => postCode.toUpperCase, postCode => postCode)
-      .verifying(emptyError(postCode), isNonEmpty)
-      .verifying(notValidError(postCode), validatePostcode(10))
+    // TODO: we need to store a postcode if provided when country non-UK. Postcode is being lost for all non-UK
+    //       addresses at the moment.
+    postCode -> mandatoryIfEqual("countryCode",
+                                 "GB",
+                                 text()
+                                   .transform[String](postCode => postCode.toUpperCase,
+                                                      postCode => postCode
+                                   )
+                                   .verifying(emptyError(postCode), isNonEmpty)
+                                   .verifying(notValidError(postCode), validatePostcode(10))
+    ),
+    countryCode -> text()
```

Comment:
We were not able, after an initial investigation, to get to the bottom of why non-uk addresses were losing their postcodes. We felt like this could be fixed at a later date rather than derailing this work.

Reply:
@SteveSugden found the answer to this one (as he coded the fix in NDRC).